server/src-lib/Hasura/RQL/Types/ScheduledTrigger.hs

```diff
@@ -166,3 +187,173 @@ instance FromJSON CreateScheduledEvent where
                                    <*> o .:? "comment"
 
 $(deriveToJSON (aesonDrop 3 snakeCase) ''CreateScheduledEvent)
+
+-- | The 'ScheduledEventType' data type is needed to differentiate
+--   between a 'CronScheduledEvent' and 'OneOffScheduledEvent' scheduled
+--   event because they both have different configurations
+--   and they live in different tables.
+data ScheduledEventType =
+    Cron
+  -- ^ A Cron scheduled event has a template defined which will
+  -- contain the webhook, header configuration, retry
+  -- configuration and a payload. Every cron event created
+  -- uses the above mentioned configurations defined in the template.
+  -- The configuration defined with the cron trigger is cached
+  -- and hence it's not fetched along the cron scheduled events.
+  | OneOff
+  -- ^ A One-off scheduled event doesn't have any template defined
+  -- so all the configuration is fetched along the scheduled events.
+    deriving (Eq, Show)
+$(deriveJSON defaultOptions{constructorTagModifier = snakeCase} ''ScheduledEventType)
+
+data ScheduledEventInvocation
+  = ScheduledEventInvocation
+  { _seiId        :: !InvocationId
+  , _seiEventId   :: !EventId
+  , _seiStatus    :: !(Maybe Int)
+  , _seiRequest   :: !(Maybe Value)
+  , _seiResponse  :: !(Maybe Value)
+  , _seiCreatedAt :: !UTCTime
+  } deriving (Show, Eq)
+$(deriveJSON (aesonDrop 4 snakeCase) ''ScheduledEventInvocation)
+
+data ScheduledEvent
+  = SEOneOff
+  | SECron !TriggerName
+  deriving (Show, Eq)
+
+parseScheduledEvent :: Object -> Parser ScheduledEvent
+parseScheduledEvent o = do
+  ty <- o .: "type"
+  case ty of
+    Cron   -> SECron <$> o .: "trigger_name"
+    OneOff -> pure SEOneOff
```

Comment:
Perhaps from a code organization point of view it makes sense to make this into a local definition as part of the `FromJSON GetEventInvocations` instance, and ditto for `parseScheduledEventPagination`, just like `parseEventId` already is?

Reply:
These types aren't necessary now. However, I'll keep these notes in mind while adding them in subsequent PRs.